src/artifact/v3/manifest/manifest.cpp

```diff
@@ -44,6 +44,7 @@ using ExpectedManifestLine = expected::expected<ManifestLine, error::Error>;
 
 const size_t expected_shasum_length {64};
 const size_t expected_whitespace {2};
+const size_t max_allowed_filename_length {4096};
```

Comment:
There are the `PATH_MAX` and `MAXPATHLEN` system macros/constants (the latter defined as the value of the former on my system). You could use one of those. But they are set to 4096 on modern GNU/Linux systems. I'd either tried doing `#include <sys/param.h>` and use one of those if available, or at least added a comment that they are most likely 4096.

Reply:
Also, should all three of those be `static`?

Reply:
I think the header and `PATH_MAX` is POSIX-only though.

Reply:
> What if for the sake of keeping it simple we just restrict it to 1000? or 100?

:+1: